### PR TITLE
Shahid/redis 6 auth issue

### PR DESCRIPTION
**Pull request by jashaik**
### Description

Added redis 6 support 

### Issues Resolved

[List any existing issues this PR resolves, or any Discourse or
StackOverflow discussions that are relevant]
in this PR

### Check List

- [ ] New functionality includes tests
- [ ] All buildkite tests pass
- [ ] Full omnibus build and tests in buildkite pass
- [ ] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/main/CONTRIBUTING.md#developer-certification-of-origin-dco>
- [ ] PR title is a worthy inclusion in the CHANGELOG

### CODE REVIEW

**Comment by jashaik** [comment]
https://buildkite.com/chef/chef-chef-server-main-omnibus-adhoc/builds/4528
https://buildkite.com/chef/chef-umbrella-main-chef-server/builds/1315

**Review by vinay-satish — APPROVED** [review]
LGTM!

File: omnibus/files/server-ctl-cookbooks/infra-server/templates/default/sv-redis_lb-run.erb [hunk 1]
```diff
@@ -1,3 +1,4 @@
...
-exec /opt/<%= ChefUtils::Dist::Org::LEGACY_CONF_DIR %>/embedded/bin/veil-env-helper -f /etc/<%= ChefUtils::Dist::Org::LEGACY_CONF_DIR %>/private-<%= ChefUtils::Dist::Infra::SHORT %>-secrets.json -s REDIS_PASSWORD=redis_lb.password -- chpst -P -o 131071 -u <%= node["private_chef"]["user"]["username"] %> -U <%= node["private_chef"]["user"]["username"] %> -o 100000 env HOME="<%= node["private_chef"]["redis_lb"]["dir"] %>" /opt/<%= ChefUtils::Dist::Org::LEGACY_CONF_DIR %>/embedded/bin/redis-server <%= File.join(node["private_chef"]["redis_lb"]["dir"], "etc", "redis.conf") %>
+exec /opt/<%= ChefUtils::Dist::Org::LEGACY_CONF_DIR %>/embedded/bin/veil-env-helper -f /etc/<%= ChefUtils::Dist::Org::LEGACY_CONF_DIR %>/private-<%= ChefUtils::Dist::Infra::SHORT %>-secrets.json -s REDIS_PASSWORD=redis_lb.password -- chpst -P -o 131071 -u <%= node["private_chef"]["user"]["username"] %> -U <%= node["private_chef"]["user"]["username"] %> -o 100000 env HOME="<%= node["private_chef"]["redis_lb"]["dir"] %>" /opt/<%= ChefUtils::Dist::Org::LEGACY_CONF_DIR %>/embedded/bin/redis-server <%= File.join(node["private_chef"]["redis_lb"]["dir"], "etc", "redis.conf") %> --requirepass "${REDIS_PASSWORD}"
```
Comment (PrajaktaPurohit) [inline omnibus/files/server-ctl-cookbooks/infra-server/templates/default/sv-redis_lb-run.erb:3]: Was the REDIS_PASSWORD passed to this script previously but never used?

Comment (jashaik) [reply omnibus/files/server-ctl-cookbooks/infra-server/templates/default/sv-redis_lb-run.erb:3]: In redis 5, The password is getting set implicitly but not happening in redis 6. So using extra argument `requirepass` while start the redis 6.
There are some major changes in Redis auth module in version 6. From redis 6 we need to explicitly set password for the default user if we want to enable external access due to ACL's.
For backward compatibility they enabled some user 'default' which should be passed when ever we use Redis.new and password should also be set if we want to access externally.
Reference - https://redis.io/docs/manual/security/acl/

**Comment by PrajaktaPurohit** [comment]
If maintenance mode continues to work and there are no chanages to what a user needs to do, I am good with merging.

**Pull request merged**